packages/video_player_videohole/tizen/src/video_player.cc

```diff
@@ -0,0 +1,582 @@
+                get_error_message(ret));
+      event_sink_->Error("player_get_video_size failed",
+                         get_error_message(ret));
+      return;
+    }
+    LOG_INFO("[VideoPlayer] video width: %d, video height: %d", width, height);
+
+    player_display_rotation_e rotation;
+    ret = player_get_display_rotation(player_, &rotation);
+    if (ret != PLAYER_ERROR_NONE) {
+      LOG_ERROR("[VideoPlayer] player_get_display_rotation failed: %s",
+                get_error_message(ret));
+      event_sink_->Error("player_get_display_rotation failed",
+                         get_error_message(ret));
+    } else {
+      if (rotation == player_display_rotation_e::PLAYER_DISPLAY_ROTATION_90 ||
+          rotation == player_display_rotation_e::PLAYER_DISPLAY_ROTATION_270) {
+        int tmp = width;
+        width = height;
+        height = tmp;
+      }
+    }
+
+    is_initialized_ = true;
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("initialized")},
+        {flutter::EncodableValue("duration"),
+         flutter::EncodableValue(duration)},
+        {flutter::EncodableValue("width"), flutter::EncodableValue(width)},
+        {flutter::EncodableValue("height"), flutter::EncodableValue(height)}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send initialized event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendBufferingStart() {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingStart")}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send bufferingStart event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendBufferingUpdate(int position) {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingUpdate")},
+        {flutter::EncodableValue("values"), flutter::EncodableValue(position)}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send bufferingUpdate event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendBufferingEnd() {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("bufferingEnd")}};
+    flutter::EncodableValue eventValue(encodables);
+    LOG_INFO("[VideoPlayer] send bufferingEnd event");
+    event_sink_->Success(eventValue);
+  }
+}
+
+void VideoPlayer::SendSubtitleUpdate(int duration, char *text) {
+  if (event_sink_) {
+    flutter::EncodableMap encodables = {
+        {flutter::EncodableValue("event"),
+         flutter::EncodableValue("subtitleUpdate")},
+        {flutter::EncodableValue("duration"),
+         flutter::EncodableValue(duration)},
```

Comment:
Does the `duration` always have a valid value greater than 0? I'm going to refactor the Dart part a bit. Which stream did you use to test the subtitle feature?

Reply:
`duration` is greater than or equal to 0. If there is subtitle text, `duration` will be greater than 0. If subtitle text is null, durantion will be 0. 
I use these streams to test: 
dash:
https://dash.akamaized.net/akamai/test/caption_test/ElephantsDream/elephants_dream_480p_heaac5_1_https.mpd
mp4:
http://109.123.121.45/test/TTXT_test_01.mp4

Reply:
Thanks, but the second URL doesn't work (the video doesn't load). I tried the first URL in the "Dash" tab (with a `ClosedCaption` widget added) but I couldn't see any caption on the screen. What could have gone wrong?

Reply:
Have you tried the second URL on 7.0? This subtitle only support on Tizen 7.0

Reply:
Oh, my TV is running Tizen 6.5.

Is it common for streams to have embedded subtitles? Wouldn't it be better to support only file-based captioning to achieve consistent behavior across all Tizen versions, just like the original `video_player` plugin?

Reply:
The `player_set_subtitle_updated_cb` function was added in Tizen 2.3, but was not supported until Tizen 7.0?

Reply:
`player_set_subtitle_updated_cb` support all. This subtitle not support , is because the subtitle format in the stream---dash sidecar only support till Tizen 7.0. However, other subtitles(not dash sidecar) are supported.
Only using file-based caption is good, but the 3rd-party want to use stream internal subtitle.

Reply:
Okay. I'm just afraid that it might embarrass the developers when they realize that the functionality doesn't work under certain conditions (Tizen 6.0-6.5 + Dash). Do you think it's not that critical?

Reply:
But it is difficult for us to fix because this is based on MMplayer Api, otherwise maybe we can remove this function and add limitation that only support file-based caption.

Reply:
Okay, thanks. Let me take care (maybe I will add the information to README).